.github/workflows/code-coverage-report.yml

```
@@ -0,0 +1,61 @@
+name: Code coverage report
+
+on:
+  workflow_call:
+    secrets:
+      GH_ACCESS_TOKEN:
+        required: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          path: project
+
+      - name: Clone gradle-scripts
+        uses: actions/checkout@v2
+        with:
+          repository: tastyworks/gradle-scripts
+          ref: 'refs/heads/master'
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+          path: gradle-scripts
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x ./project/gradlew
```

you can use this existing setup action which does java/gradle setup

https://github.com/tastyworks/boulangerie-shared-github-workflows/blob/main/.github/common-setup/action.yml

```suggestion
      - uses: tastyworks/boulangerie-shared-github-workflows/.github/common-setup@main
        with:
          GH_ACCESS_TOKEN: ${{ secrets.GH_ACCESS_TOKEN }}
```